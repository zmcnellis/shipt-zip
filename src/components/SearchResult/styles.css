/* Search results ------------------------------------------------------------*/
.results-container {
  display: flex;
  flex-direction: column;
  width: 100%;
  margin: 0 auto;
  max-width: 1080px;
  justify-content: space-around;
}

.results-header {
  text-align: center;
  text-transform: uppercase;
  font-size: 11px;
}

.search-result {
  text-align: center;
  flex-grow: 1;
  flex-basis: 0;
  width: auto;
  padding: 0 0 20px 0;
}

.store-name {
  font-size: 28px;
}

.launch-date {
  font-size: 11px;
  font-weight: bold;
  color: #7b7b7b;
}

.button-area {
  text-align: center;
  padding: 20px 0;
}

/* Media queries -------------------------------------------------------------*/
@media (min-width: 700px) {
  /* search results */
  .results-container {
    flex-direction: row;
  }
}
