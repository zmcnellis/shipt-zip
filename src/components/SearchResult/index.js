import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './styles.css';

class SearchResult extends Component {

  formatDate(dateStr) {
    const date = new Date(dateStr);

    if (date - new Date() <= 0) {
      return "Now Available";
    }
    else {
      return `Coming ${date.getMonth() + 1}/${date.getDate()}`;
    }
  }

  render() {
    return (
      <div className="search-result">
        <h2 className="store-name">{this.props.name}</h2>
        <p className="launch-date">{this.formatDate(this.props.date)}</p>
      </div>
    );
  }
}

SearchResult.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired
};

export default SearchResult;
