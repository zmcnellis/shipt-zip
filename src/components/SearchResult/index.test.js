import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import SearchResult from '.';

describe('SearchResult', () => {
  it('should render correctly', () => {
    const output = shallow(
      <SearchResult id="1" name="mockName" date="2016-01-29T00:00:00.000-05:00"/>
    );
    expect(output).toMatchSnapshot();
  });

  describe('should show the correct date', () => {

    it('when date is in the past', () => {
      const date = new Date();
      date.setDate(date.getDate() - 1);

      const output = shallow(
        <SearchResult id="1" name="mockName" date={date.toString()}/>
      );
      expect(output.find('.launch-date').text()).toEqual('Now Available');
    });

    it('when date is in the future', () => {
      const date = new Date();
      date.setDate(date.getDate() + 1);

      const output = shallow(
        <SearchResult id="1" name="mockName" date={date.toString()}/>
      );

      expect(output.find('.launch-date').text()).toEqual(`Coming ${date.getMonth() + 1}/${date.getDate()}`);
    });

    it('when date is equal to the current date', () => {
      const date = new Date();

      const output = shallow(
        <SearchResult id="1" name="mockName" date={date.toString()}/>
      );
      expect(output.find('.launch-date').text()).toEqual('Now Available');
    });
  });
});
