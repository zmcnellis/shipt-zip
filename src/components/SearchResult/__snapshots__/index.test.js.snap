// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResult
    date="2016-01-29T00:00:00.000-05:00"
    id="1"
    name="mockName"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="store-name"
>
          mockName
</h2>,
        <p
          className="launch-date"
>
          Now Available
</p>,
      ],
      "className": "search-result",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mockName",
          "className": "store-name",
        },
        "ref": null,
        "rendered": "mockName",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Now Available",
          "className": "launch-date",
        },
        "ref": null,
        "rendered": "Now Available",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="store-name"
>
            mockName
</h2>,
          <p
            className="launch-date"
>
            Now Available
</p>,
        ],
        "className": "search-result",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mockName",
            "className": "store-name",
          },
          "ref": null,
          "rendered": "mockName",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Now Available",
            "className": "launch-date",
          },
          "ref": null,
          "rendered": "Now Available",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
